### PR TITLE
Rename Preload attribute to LoadableDescriptors

### PR DESCRIPTION
**Pull request by theresa-m**
Fixes: https://github.com/eclipse-openj9/openj9/issues/19611

I tried to split commits as follows for easier reviewing:
1. Test changes
2. VM logic changes
3. VM renaming and formatting

### CODE REVIEW

**Comment by keithc-ca** [comment]
Please address the line endings check failure.

File: debugtools/DDR_VM/src/com/ibm/j9ddr/CompatibilityConstants29.dat
Comment (keithc-ca) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/CompatibilityConstants29.dat:88]: In general, renaming a constant would create problems accessing system dumps where the old name is used, but given this is for a preview feature, this is, perhaps, acceptable. Please do keep the names sorted (swap lines 87 & 88).

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/OptInfo.java
Comment (keithc-ca) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/OptInfo.java:31]: Please keep imports sorted.

File: runtime/cfdumper/main.c
Comment (keithc-ca) [inline runtime/cfdumper/main.c:935]: Please add the missing space after `for`, here and on line 937.

File: runtime/nls/cfre/cfrerr.nls
Comment (keithc-ca) [inline runtime/nls/cfre/cfrerr.nls:1830]: Suggest "Descriptor" should be capitalized.

File: runtime/oti/util_api.h
Comment (keithc-ca) [inline runtime/oti/util_api.h:1362]: Please remove the trailing period (compare with line 1374).

File: runtime/util/optinfo.c
Comment (keithc-ca) [inline runtime/util/optinfo.c:732]: Please finish the sentence with a period.

File: runtime/util/rcdump.c
Comment (keithc-ca) [inline runtime/util/rcdump.c:83]: Please remove the space before `(` (compare with line 894).

File: runtime/vm/createramclass.cpp
Comment (keithc-ca) [inline runtime/vm/createramclass.cpp:2050]: Please finish the sentence with a period; also line 2054.

File: runtime/vm/createramclass.cpp
Comment (keithc-ca) [inline runtime/vm/createramclass.cpp:2057]: Suggest this use `-=` for clarity; without blank line 2057:
```c
				loadableDescriptorLength -= 2;
```

File: test/functional/Valhalla/src/org/openj9/test/lworld/ValhallaAttributeTests.java
Comment (keithc-ca) [inline test/functional/Valhalla/src/org/openj9/test/lworld/ValhallaAttributeTests.java:31]: Suggest this class could be `static`.

File: test/functional/Valhalla/src/org/openj9/test/lworld/ValhallaAttributeTests.java
Comment (keithc-ca) [inline test/functional/Valhalla/src/org/openj9/test/lworld/ValhallaAttributeTests.java:41]: One argument per line, please:
```java
		ValhallaAttributeGenerator.generateClassWithTwoLoadableDescriptorsAttributes(
			"MultiLoadableDescriptorsAttributes",
			new String[] { "L" + className + ";" },
			new String[] { "L" + className + ";" }
		);
```

File: test/functional/Valhalla/src/org/openj9/test/lworld/ValhallaAttributeTests.java
Comment (keithc-ca) [inline test/functional/Valhalla/src/org/openj9/test/lworld/ValhallaAttributeTests.java:44]: `static`?

File: test/functional/Valhalla/src/org/openj9/test/lworld/ValhallaUtils.java
Comment (keithc-ca) [inline test/functional/Valhalla/src/org/openj9/test/lworld/ValhallaUtils.java:42]: Please list modifiers in canonical order:
```java
	static final class LoadableDescriptorsAttribute extends Attribute {
```
Please do the same for other nested classes.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: runtime/vm/createramclass.cpp
Comment (keithc-ca) [inline runtime/vm/createramclass.cpp:2057]: This blank line is not helpful, please remove it.

File: test/functional/Valhalla/src/org/openj9/test/lworld/ValhallaAttributeTests.java
Comment (keithc-ca) [inline test/functional/Valhalla/src/org/openj9/test/lworld/ValhallaAttributeTests.java:50]: Please finish each sentence with a period (at least those lines added or modified by this change).

File: test/functional/Valhalla/src/org/openj9/test/lworld/ValhallaAttributeTests.java
Comment (keithc-ca) [inline test/functional/Valhalla/src/org/openj9/test/lworld/ValhallaAttributeTests.java:36]: Please list modifiers in canonical order (`public static void ...`) (at least those declarations added or modified by this change).

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: test/functional/Valhalla/src/org/openj9/test/lworld/ValhallaAttributeTests.java
Comment (keithc-ca) [reply test/functional/Valhalla/src/org/openj9/test/lworld/ValhallaAttributeTests.java]: This does not appear to have been addressed.

File: test/functional/Valhalla/src/org/openj9/test/lworld/ValhallaAttributeTests.java
Comment (theresa-m) [inline test/functional/Valhalla/src/org/openj9/test/lworld/ValhallaAttributeTests.java:36]: @keithc-ca the order is switched here and in other uses in this file.

**Comment by keithc-ca** [comment]
Jenkins test sanity aarch64_linux,aarch64_linux_vt_standard jdknext

**Comment by keithc-ca** [comment]
Updates are required to https://github.com/ibmruntimes/openj9-openjdk-jdk.valuetypes before that repository will compile with the current content of https://github.com/eclipse-openj9/openj9:
```
[2024-10-17T16:51:51.110Z] /home/jenkins/workspace/Build_JDKnext_aarch64_linux_vt_standard_Personal/build/linux-aarch64-server-release/support/j9jcl/jdk.management/share/classes/com/ibm/lang/management/internal/PlatformMBeanProvider.java:40: error: cannot find symbol
[2024-10-17T16:51:51.110Z] import com.sun.management.internal.VirtualThreadSchedulerImpls;
[2024-10-17T16:51:51.110Z]                                   ^
[2024-10-17T16:51:51.110Z]   symbol:   class VirtualThreadSchedulerImpls
[2024-10-17T16:51:51.110Z]   location: package com.sun.management.internal
[2024-10-17T16:51:51.110Z] /home/jenkins/workspace/Build_JDKnext_aarch64_linux_vt_standard_Personal/build/linux-aarch64-server-release/support/j9jcl/jdk.management/share/classes/com/ibm/lang/management/internal/PlatformMBeanProvider.java:41: error: package jdk.management does not exist
[2024-10-17T16:51:51.110Z] import jdk.management.VirtualThreadSchedulerMXBean;
[2024-10-17T16:51:51.110Z]                      ^
[2024-10-17T16:51:51.110Z] /home/jenkins/workspace/Build_JDKnext_aarch64_linux_vt_standard_Personal/build/linux-aarch64-server-release/support/j9jcl/jdk.management/share/classes/com/ibm/lang/management/internal/PlatformMBeanProvider.java:114: error: cannot find symbol
[2024-10-17T16:51:51.110Z] 			.addInterface(VirtualThreadSchedulerMXBean.class)
[2024-10-17T16:51:51.110Z] 			              ^
[2024-10-17T16:51:51.110Z]   symbol:   class VirtualThreadSchedulerMXBean
[2024-10-17T16:51:51.110Z]   location: class PlatformMBeanProvider
[2024-10-17T16:51:51.110Z] /home/jenkins/workspace/Build_JDKnext_aarch64_linux_vt_standard_Personal/build/linux-aarch64-server-release/support/j9jcl/jdk.management/share/classes/com/ibm/lang/management/internal/PlatformMBeanProvider.java:113: error: cannot find symbol
[2024-10-17T16:51:51.110Z] 		ComponentBuilder.create("jdk.management:type=VirtualThreadScheduler", VirtualThreadSchedulerImpls.create()) //$NON-NLS-1$
[2024-10-17T16:51:51.110Z] 		                                                                      ^
[2024-10-17T16:51:51.110Z]   symbol:   variable VirtualThreadSchedulerImpls
[2024-10-17T16:51:51.110Z]   location: class PlatformMBeanProvider
[2024-10-17T16:51:51.110Z] 4 errors
```

**Comment by keithc-ca** [comment]
I created https://github.com/eclipse-openj9/openj9/pull/20377 which should unblock this.

**Comment by pshipton** [comment]
Jenkins test sanity aarch64_linux_vt_standard jdknext

**Comment by keithc-ca** [comment]
There are many failures like this:
```
FAILED CONFIGURATION: @BeforeMethod setUp
java.lang.InternalError: java.lang.IllegalAccessException: static final field has no write access: jdk.internal.misc.Unsafe.theUnsafe/jdk.internal.misc.Unsafe/putStatic, from class java.lang.Object (module java.base)
	at java.base/jdk.internal.reflect.MethodHandleAccessorFactory.newFieldAccessor(MethodHandleAccessorFactory.java:207)
	at java.base/jdk.internal.reflect.ReflectionFactory.newFieldAccessor(ReflectionFactory.java:144)
	at java.base/java.lang.reflect.Field.acquireOverrideFieldAccessor(Field.java:1212)
	at java.base/java.lang.reflect.Field.getOverrideFieldAccessor(Field.java:1181)
	at java.base/java.lang.reflect.Field.get(Field.java:454)
	at org.openj9.test.unsafe.UnsafeTestBase.getUnsafeInstance(UnsafeTestBase.java:120)
	at org.openj9.test.unsafe.UnsafeTestBase.setUp(UnsafeTestBase.java:562)
	[...]
Caused by: java.lang.IllegalAccessException: static final field has no write access: jdk.internal.misc.Unsafe.theUnsafe/jdk.internal.misc.Unsafe/putStatic, from class java.lang.Object (module java.base)
	at java.base/java.lang.invoke.MemberName.makeAccessException(MemberName.java:913)
	at java.base/java.lang.invoke.MethodHandles$Lookup.unreflectField(MethodHandles.java:3608)
	at java.base/java.lang.invoke.MethodHandles$Lookup.unreflectSetter(MethodHandles.java:3599)
	at java.base/java.lang.invoke.MethodHandleImpl$1.unreflectField(MethodHandleImpl.java:1621)
	at java.base/jdk.internal.reflect.MethodHandleAccessorFactory.newFieldAccessor(MethodHandleAccessorFactory.java:185)
	... 31 more
```

**Comment by theresa-m** [comment]
Those failures are related to https://github.com/eclipse-openj9/openj9/issues/20372

**Comment by hangshao0** [comment]
Do you have more review comments on this ? @keithc-ca

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
> Do you have more review comments on this?

No more comments other than to say I'd be happier if we didn't have to ignore test failures, but I'll defer to you, @hangshao0.

File: runtime/nls/cfre/cfrerr.nls
Comment (hangshao0) [inline runtime/nls/cfre/cfrerr.nls:1816]: We could mention there should be only one LoadableDescriptors attribute either in the NLS message or in the explanation so that it is more obvious to the users what is wrong.

**Comment by hangshao0** [comment]
Jenkins test sanity.functional,extended alinuxval jdknext

**Review by hangshao0 — APPROVED** [review]

**Pull request merged**
